Source/MediaInfo/Tag/File_Exif.cpp

```diff
@@ -815,56 +984,347 @@ void File_Exif::Streams_Finish()
+            case IFDExif::SubSecTime:
+            case IFDExif::SubSecTimeOriginal:
+            case IFDExif::SubSecTimeDigitized:
+                Parameter = (size_t)-1;
+                break;
+            case IFDExif::DateTimeOriginal:
+                Parameter = General_Recorded_Date;
+                Value = MergeDateTimeSubSecOffset(Item.second, IFDExif::SubSecTimeOriginal, IFDExif::OffsetTimeOriginal);
+                break;
+            case IFDExif::DateTimeDigitized:
+                Parameter = General_Encoded_Date;
+                Value = MergeDateTimeSubSecOffset(Item.second, IFDExif::SubSecTimeDigitized, IFDExif::OffsetTimeDigitized);
+                break;
+            case IFDExif::FlashpixVersion: {
+                ParameterC = "FlashpixVersion";
+                string flashpix_ver{ Item.second.Read().To_UTF8() };
+                if (flashpix_ver.size() == 4) {
+                    flashpix_ver.insert(2, 1, '.');
+                    if (flashpix_ver.front() == '0')
+                        flashpix_ver.erase(0, 1);
+                }
+                Value.From_UTF8(flashpix_ver);
+                break;
+            }
+            case IFDExif::WhiteBalance: ParameterC = "AutoWhiteBalanceMode"; Value = Exif_IFDExif_WhiteBalance_Name(Item.second.Read().To_int16u()); break;
+            }
+            FillMetadata(Value, Item, Parameter, ParameterC, ParameterS);
+            switch (Item.first) {
+            case IFDExif::ExifVersion:
+            case IFDExif::FlashpixVersion: Fill_SetOptions(Stream_General, 0, ParameterC, "N NF"); break;
+            }
+        }
     }
+
     if (Infos_GPS_It != Infos.end()) {
-        const auto Infos_GPS = Infos_GPS_It->second;
-        const auto GPSLatitude = Infos_GPS.find(IFDGPS::GPSLatitude);
-        const auto GPSLatitudeRef = Infos_GPS.find(IFDGPS::GPSLatitudeRef);
-        const auto GPSLongitude = Infos_GPS.find(IFDGPS::GPSLongitude);
-        const auto GPSLongitudeRef = Infos_GPS.find(IFDGPS::GPSLongitudeRef);
-        const auto GPSAltitude = Infos_GPS.find(IFDGPS::GPSAltitude);
-        if (GPSLatitude != Infos_GPS.end() && GPSLatitude->second.size() == 3 && GPSLatitudeRef != Infos_GPS.end() && GPSLongitude != Infos_GPS.end() && GPSLongitude->second.size() == 3 && GPSLongitudeRef != Infos_GPS.end()) {
-            Ztring location = GPSLatitude->second.at(0) + Ztring().From_UTF8("\xC2\xB0");
-            if (GPSLatitude->second.at(1) != __T("0") || GPSLatitude->second.at(0).find(__T('.')) == string::npos) {
-                location += GPSLatitude->second.at(1) + __T('\'');
+        currentIFD = Kind_GPS;
+        const auto& Infos_GPS = Infos_GPS_It->second;
+        for (const auto& Item : Infos_GPS) {
+            size_t Parameter = 0;
+            Ztring Value;
+            const char* ParameterC = nullptr;
+            string ParameterS;
+            switch (Item.first) {
+            case IFDGPS::GPSVersionID: {
+                ParameterC = "GPSVersionID";
```

Comment:
```suggestion
                ParameterC = "ExifGPSVersion";
```

ID is not really relevant, and we need to indicate that this is from Exif.